internal/directives/helm_chart_updater.go

```diff
@@ -160,20 +160,26 @@ func (h *helmChartUpdater) processChartUpdates(
 			stepCtx.Freight.References(),
 			repoURL,
 			chartName,
+			true, // Treat as an error if not found
 		)
 		if err != nil {
 			return nil, fmt.Errorf("failed to find chart: %w", err)
 		}
-		if chart == nil {
-			continue
-		}
 
+		var updateUsed bool
 		for i, dep := range chartDependencies {
 			if dep.Repository == update.Repository && dep.Name == update.Name {
 				changes[fmt.Sprintf("dependencies.%d.version", i)] = chart.Version
+				updateUsed = true
 				break
 			}
 		}
+		if !updateUsed {
+			return nil, fmt.Errorf(
+				"no dependency in Chart.yaml matched update with repository %s and name %q",
+				update.Repository, update.Name,
+			)
+		}
```

Comment:
If we looped through all the dependencies and found none that matched the update, this is evidence of misconfiguration.